### PR TITLE
feat: add slim image

### PR DESCRIPTION
**Pull request by sergelogvinov**
#78

Build two images full size and slim.

```shell
ghcr.io/sergelogvinov/kubelet   v1.31.0-rc.1-slim   c925dc6995c0   20 minutes ago   211MB
ghcr.io/sergelogvinov/kubelet   v1.31.0-rc.1        c129faaaf339   21 minutes ago   411MB
```

What do you think to build two images?

### CODE REVIEW

**Comment by smira** [comment]
Yes, I think reducing kubelet image size is a great step forward.

With Kubernetes removing in-tree CSI drivers, I'm not sure if e.g. `ceph-common` makes any sense for modern kubelet (i.e. 1.31+). 

But your PR might be a good way forward to first introduce `-slim` images, get them tested and start removing stuff from default images.

File: Dockerfile [hunk 1]
```diff
@@ -55,6 +57,25 @@ COPY --from=builder /kubelet /usr/local/bin/kubelet
...
+########################
+
+FROM registry.k8s.io/build-image/debian-iptables:bookworm-v1.0.0 AS container-slim
+
+RUN clean-install \
+  --allow-change-held-packages \
+  ca-certificates \
+  libcap2 \
+  ethtool \
+  iproute2 \
```
Comment (smira) [inline Dockerfile:73]: socat should be small and seems to be listed as a dependency in upstream kubelet package

Comment (smira) [reply Dockerfile:73]: https://github.com/kubernetes/release/blob/ea923dd26f66ccc610f865cfa8a3705dccf153c8/cmd/krel/templates/latest/kubelet/kubelet.spec#L23-L33

Comment (sergelogvinov) [reply Dockerfile:73]: I remember the time when socat was required for kubectl port-forward, i've tested it recently it works without socat.
But we can keep it as well. 

I believe, that most of the tools are legacy. And kubernetes community do not want to spent time to removal process yet. 

Also my research https://github.com/kubernetes/mount-utils/pull/17

Comment (frezbo) [reply Dockerfile:73]: > I remember the time when socat was required for kubectl port-forward, i've tested it recently it works without socat.

should we keep it them @smira ?

Comment (sergelogvinov) [reply Dockerfile:73]: i've added `socat` to slim version...

Comment (smira) [reply Dockerfile:73]: What I'm thinking is what if for K8s 1.31+ we offer slim version by default and do a :scream: test?

Comment (frezbo) [reply Dockerfile:73]: yeh, and still keep the old one around, so if something breaks it's a machine config patch to fix

Comment (smira) [reply Dockerfile:73]: The previous comment was more of a question actually

**Comment by smira** [comment]
I will refactor this a bit to support what we've discussed so far

**Comment by smira** [comment]
@frezbo @sergelogvinov please let me know what do you think about the changes I made (in a separate commit)

**Review by frezbo — APPROVED** [review]

**Pull request merged**